src/main/java/uk/nhs/digital/nhsconnect/nhais/controller/AmendmentController.java

```diff
@@ -1,29 +1,96 @@
+import uk.nhs.digital.nhsconnect.nhais.model.mesh.OutboundMeshMessage;
+import uk.nhs.digital.nhsconnect.nhais.service.JsonPatchToEdifactService;
+import uk.nhs.digital.nhsconnect.nhais.service.OutboundQueueService;
+import uk.nhs.digital.nhsconnect.nhais.utils.HttpHeaders;
 
+import java.io.IOException;
+import java.util.Collections;
+import java.util.stream.Collectors;
 
 @RestController
 @Slf4j
 @RequiredArgsConstructor(onConstructor = @__(@Autowired))
 public class AmendmentController {
 
+    private final JsonPatchToEdifactService jsonPatchToEdifactService;
+    private final OutboundQueueService outboundQueueService;
+    private final ObjectMapper objectMapper;
+
     @PatchMapping(path= "/fhir/Patient/{nhsNumber}", consumes="application/json", produces="application/json")
     @ResponseStatus(HttpStatus.ACCEPTED)
-    public ResponseEntity<?> amendment(@PathVariable(name= "nhsNumber") String nhsNumber, @RequestBody AmendmentBody amendmentBody) {
+    public ResponseEntity<?> amendment(@PathVariable(name= "nhsNumber") String nhsNumber, @RequestBody String body) {
+        AmendmentBody amendmentBody = parseRequest(body);
         LOGGER.info("Amendment request: {}", amendmentBody);
-        return new ResponseEntity<>(HttpStatus.ACCEPTED);
+        validateRequest(nhsNumber, amendmentBody);
+        OutboundMeshMessage meshMessage = jsonPatchToEdifactService.convertToEdifact(amendmentBody);
+        outboundQueueService.publish(meshMessage);
+        MultiValueMap<String, String> headers = new LinkedMultiValueMap<>();
+        headers.put(HttpHeaders.OPERATION_ID, Collections.singletonList(meshMessage.getOperationId()));
+        return new ResponseEntity<>(headers, HttpStatus.ACCEPTED);
+    }
+
+    private AmendmentBody parseRequest(@RequestBody String body) {
+        try {
+            JsonParser parser = objectMapper.createParser(body);
+            return parser.readValueAs(AmendmentBody.class);
+        } catch (IOException ex) {
+            throw new AmendmentValidationException(ex.getMessage());
+        }
+    }
+
+    private void validateRequest(String nhsNumber, AmendmentBody amendmentBody) {
+        validateRequiredFields(amendmentBody);
+
+        if (!nhsNumber.equals(amendmentBody.getNhsNumber())) {
+            throw new AmendmentValidationException("Request body has different NHS number than provided in request path");
+        }
+
+        var amendmentPaths = amendmentBody.getPatches().stream()
+                .map(AmendmentPatch::getPath)
+                .collect(Collectors.toSet());
+
+        if (amendmentBody.getPatches().size() != amendmentPaths.size()) {
+            throw new AmendmentValidationException("Request contains path that is used multiple times. Each patch path must only be used once within the amendment request");
+        }
+    }
+
+    private void validateRequiredFields(AmendmentBody amendmentBody) {
+        if (amendmentBody.getNhsNumber() == null) {
```

Comment:
Could we use isBlank instead of a null check? It doesn't look this this story defines missing, but our other stories define it as
“missing” is any field that is:

- not present
- null, empty, or only contains whitespace

Reply:
Doing the validation here is much cleaner and more straightforward than waiting (hoping?) for something further downstream to pick up the error

Reply:
Changed to `StringUtils.isEmpty()`